enhancements/kube-apiserver/bound-sa-tokens.md

```diff
@@ -0,0 +1,146 @@
+---
+title: bound-service-account-tokens
+authors:
+  - "@marun"
+reviewers:
+  - "@deads2k"
+  - "@mfojtik"
+  - "@stlaz"
+  - "@sttts"
+approvers:
+  - TBD
+creation-date: 2019-11-28
+last-updated: 2019-11-28
+status: provisional
+see-also:
+  - "https://github.com/kubernetes/community/blob/master/contributors/design-proposals/auth/bound-service-account-tokens.md"
+  - "https://docs.google.com/document/d/1XcOsEv4jO9P1QQHn-tOnC80oMyCm85hGA6LqHRfjTgo/edit?ts=5ddb86c1"
+  - "https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/"
+  - "https://thenewstack.io/no-more-forever-tokens-changes-in-identity-management-for-kubernetes/"
+  - "https://jpweber.io/blog/a-look-at-tokenrequest-api/"
+replaces:
+superseded-by:
+---
+
+# Bound Service Account Tokens
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+- Is it necessary/desirable to allow `service-account-max-token-expiration` to be configured?
+
+## Summary
+
+Enable the optional use of bound service account tokens via volume projection and the
+TokenRequest API.
+
+## Motivation
+
+3rd party IAM components such as the [AWS pod identity
+webhook](https://github.com/aws/amazon-eks-pod-identity-webhook) require bound tokens to
+be able to identify pods without the use of workarounds such as proxies. It is also
+generally desirable to support bound tokens to limit the scope of permissions (and
+therefore risk of compromise) for a given service account token.
+
+### Goals
+
+- A pod can request a bound service account token via a projected volume.
+- A bound token can be acquired via the TokenRequest API.
+- The public key used to validate bound tokens can be retrieved for use with 3rd party IAM
+  components.
+
+### Non-Goals
+
+- Integrate or ship the aws pod identity webhook
```

Comment:
@derekwaynecarr @cuppett do we get to punt or not?

Reply:
According to the [epic](https://jira.coreos.com/browse/MSTR-893) this is a non-goal for this team and this release.

Reply:
I was under the assumption that one of the exit criteria would be to make sure the iam setup works.. I don't think we have to ship it, but we have to verify it works with _sane_ steps.

Reply:
This is called out in the testing section.